classes/class-homepages.php

```diff
@@ -203,7 +203,7 @@ public function set_404_on_pagination() {
 	 */
 	public function get_latest_homepage_id() {
 		// Get the previewed homepage.
-		if ( is_preview() && isset( $_GET['p'] ) ) {
+		if ( is_preview() && isset( $_GET['p'] ) && isset( $_GET['preview_nonce'] ) && wp_verify_nonce( sanitize_key( $_GET['preview_nonce'] ), 'post_preview_' . absint( $_GET['p'] ) ) ) {
```

Comment:
I don't feel great about this line because I'm uncertain if this 'fix' for phpstan/phpcs is logically appropriate

Reply:
I think this is likely to break things. Testing this, I don't see `preview_nonce` in the query parameters, so it will be unset here, which will break the logic. I would suggest reverting this change.

If we're concerned about security here, for whatever reason, we may want to look at why the preview is being filtered the way it is.